frame.js

```diff
@@ -0,0 +1,70 @@
+class Frame {
+  constructor(round = 1) {
+    this.points = 0;
+    this.round = round;
+    this.rollOne = null;
+    this.rollTwo = null;
+    this.rollThree = null;
+    this.complete = false;
+  };
+
+  roll(pins) {
+    if (this.complete) throw "It's over! Stop lobbing the balls!";
+  
+    if (this.rollOne === null) {
+      this.firstGo(pins);
+    } else if (this.round < 10 && this.rollTwo === null) {
+      this.secondGo(pins);
+    } else if (this.rollTwo === null) {
+      this.secondGoRound10(pins);
+    } else {
+      this.thirdGo(pins);
+    }
+  }
+
+  strike() {
+    return this.rollOne === 10;
+  }
+  
+  spare() {
+    return this.rollOne !== 10 && this.rollOne + this.rollTwo === 10;
+  }
```

Comment:
These seem to be predicate methods - it might be worth naming them isSpare() and isStrike() as I believe this is convention (please don't ask me to source that)